### PR TITLE
feat(framework:skip) Update Docker Compose `complete` files for 1.13.0

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

Updates the Docker Compose complete setup to incorporate the upcoming 1.13.0 changes.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/docker/complete/compose.yml [hunk 1]
```diff
@@ -4,13 +4,19 @@ services:
     image: flwr/superlink:${FLWR_VERSION:-1.12.0}
```
Comment (chongshenng) [inline src/docker/complete/compose.yml:4]:
```suggestion
    image: flwr/superlink:${FLWR_VERSION:-1.13.0}
```

File: src/docker/complete/compose.yml [hunk 2]
```diff
@@ -4,13 +4,19 @@ services:
...
 
-  # create a SuperExec service
-  superexec:
+  # create a ServerApp service
+  serverapp:
     build:
       context: ${PROJECT_DIR:-.}
       dockerfile_inline: |
-        FROM flwr/superexec:${FLWR_VERSION:-1.12.0}
+        FROM flwr/serverapp:${FLWR_VERSION:-1.12.0}
```
Comment (chongshenng) [inline src/docker/complete/compose.yml:19]:
```suggestion
        FROM flwr/serverapp:${FLWR_VERSION:-1.13.0}
```

**Review by chongshenng — CHANGES_REQUESTED** [review]
@Robert-Steiner Thanks for this. I think the only the Docker image tags needs to be changed to `nightly` since it doesn't work with `1.12` right now.

File: src/docker/complete/compose.yml (hunk 1 above)
Comment (Robert-Steiner) [reply src/docker/complete/compose.yml:4]: good point!

**Comment by Robert-Steiner** [comment]
> @Robert-Steiner Thanks for this. I think the only the Docker image tags needs to be changed to `nightly` since it doesn't work with `1.12` right now.

I changed it to nightly, but we’ll need to update it to 1.13.0 before releasing 1.13.0, to make it part of the release tag. I used to change the version after the release, but this caused an issue with the link in the [Run Flower Quickstart Examples with Docker Compose](https://flower.ai/docs/framework/docker/run-quickstart-examples-docker-compose.html) doc, where the link points to the correct tag, but the default version is still set to the previous release:

```
curl https://raw.githubusercontent.com/adap/flower/refs/tags/v1.12.0/src/docker/complete/compose.yml \
    -o compose.yml
```

I’ve added a note in the release docs.

**Review by chongshenng — APPROVED** [review]
lgtm!

**Comment by chongshenng** [comment]
> I changed it to nightly, but we’ll need to update it to 1.13.0 before releasing 1.13.0, to make it part of the release tag

@Robert-Steiner yes, makes sense!

**Review by tanertopal — APPROVED** [review]
lgtm!

**Pull request merged**